src/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -0,0 +1,192 @@
+        );
+      }
+
+      const isLocalWithEmbed = audioType === "file" && embedCode;
+      if (isLocalWithEmbed) {
+        // Question: should we show this now?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `embedCode` prop will be ignored when the audioType prop is set to "file".'
+        );
+      }
+      const isLocalWithoutPath = audioType === "file" && !filePath;
+      if (isLocalWithoutPath) {
+        // Question: Should we remove this warning since we don't support filepath?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `filePath` prop is required when the `audioType` prop is set to "file."'
+        );
+      }
+
+      const isThirdPartyService = !!thirdPartyServices.find(
+        (service) => service === audioType
+      );
+      const isThirdPartyAndLocal = isThirdPartyService && filePath;
+      if (isThirdPartyAndLocal) {
+        console.warn(
+          "NYPL Reservoir AudioPlayer: The `filePath` prop will be ignored when using a 3rd party streaming service."
+        );
+      }
+
+      const isThirdPartyWithoutCode = isThirdPartyService && !embedCode;
+      if (isThirdPartyWithoutCode) {
+        console.warn(
+          "NYPL Reservoir AudioPlayer: The `embedCode` prop is required when using a 3rd party streaming service."
+        );
+        isInvalidEmbed = true;
+      }
+
+      const styles = useMultiStyleConfig("AudioPlayer", {});
+
+      /**
+       * Builder function to build the finalized render element, we prioritize localfile over 3rd party code
+       * but for the current version, we only use 3rd party code.
+       */
+      function buildFinalElement(): JSX.Element {
+        return iframeDoc ? (
+          <Box
+            dangerouslySetInnerHTML={{
+              __html: iframeDoc.outerHTML,
+            }}
+          />
+        ) : null;
+      }
+
+      const embedElement = buildFinalElement();
+
+      return (
+        <Box
+          className={className}
+          data-testid="audio-player-component"
+          id={id}
+          ref={ref}
+          __css={styles}
+          {...rest}
+        >
+          <ComponentWrapper
+            headingText={headingText}
+            descriptionText={descriptionText}
+            helperText={helperText}
+            helperTextStyles={styles.helperErrorText}
+            id={`${id}-componentWrapper`}
+          >
+            {isInvalidEmbed ? (
+              <Box
+                dangerouslySetInnerHTML={{ __html: errorMessage }}
+                __css={styles.invalid}
+              />
+            ) : (
+              <>{embedElement}</>
+            )}
+          </ComponentWrapper>
+        </Box>
+      );
+    }
+  )
+);
+
+/**
+ * Parse the embedCode string to a DOM object for proper formatted HTML element.
+ * We are only interested in the iframe element
+ */
+export function parseIframeEmbedCode(
+  embedCode: string
```

Comment:
From the `AudioPlayerTypes`, the `embedCode` prop is optional and it can be undefined. So here, it should also be set to a potential undefined value `string | undefined`.